compiler/noirc_frontend/src/tests.rs

```diff
@@ -3223,6 +3237,50 @@ fn infer_globals_to_u32_from_type_use() {
         }
     "#;
 
+    let errors = get_program_errors(src);
+    assert_eq!(errors.len(), 3);
+    assert!(matches!(
+        errors[0].0,
+        CompilationError::ResolverError(ResolverError::UnspecifiedGlobalType { .. })
+    ));
+    assert!(matches!(
+        errors[1].0,
+        CompilationError::ResolverError(ResolverError::UnspecifiedGlobalType { .. })
+    ));
+    assert!(matches!(
+        errors[2].0,
+        CompilationError::ResolverError(ResolverError::UnspecifiedGlobalType { .. })
+    ));
+}
+
+#[test]
+fn infer_partial_global_types() {
+    let src = r#"
+        pub global ARRAY: [Field; _] = [0; 3];
+        pub global STR: str<_> = "hi";
+        pub global FMT_STR: fmtstr<_, _> = f"hi {ARRAY}";
+
+        fn main() { }
+    "#;
+
+    let errors = get_program_errors(src);
+    assert_eq!(errors.len(), 0);
+}
```

Comment:
A little torn on this. I can see some benefits for the code writer in the case of strings but I think that the consistency of requiring these to be fully specified is good.

Reply:
Ok, sounds good